### PR TITLE
Enable tracked_votes processing only if http chain_ro category is configured

### PR DESCRIPTION
**Pull request by linh2931**
The chain_api_plugin does extra processing to support `v1/chain/get_finalizer_info` http endpoint. If the http `chain_ro` category is not enabled then there is no need to do the extra processing.

Resolves https://github.com/AntelopeIO/spring/issues/515

### CODE REVIEW

File: plugins/chain_plugin/chain_plugin.cpp [hunk 1]
```diff
@@ -975,14 +975,39 @@ void chain_plugin_impl::plugin_initialize(const variables_map& options) {
...
+            }
+         }
+      }
+
+      // only enable last tracked votes if chain_api_plugin and chain_ro are enabled.
+      bool tracking_enabled = chain_api_plugin_configured && chain_ro_category_configured;
+      _last_tracked_votes.emplace(*chain, tracking_enabled);
+
+      // only enable last tracked votes if chain_api_plugin enabled.
       _get_info_db.emplace(*chain, chain_api_plugin_configured);
```
Comment (heifner) [inline plugins/chain_plugin/chain_plugin.cpp:1011]: `chain_apis::read_only` has a `last_tracked_votes`. Seems like `chain_api_plugin::plugin_startup` could ask `http_plugin` if `chain_ro` category is enabled and then inform `last_tracked_votes`.

**Review by heifner — CHANGES_REQUESTED** [review]

File: plugins/chain_plugin/chain_plugin.cpp (hunk 1 above)
Comment (linh2931) [reply plugins/chain_plugin/chain_plugin.cpp:1011]: Thanks! This is a good idea. 

I was trying to query `http_plugin` from here but did not like introducing dependency of `chain_plugin` on `http_plugin`. I will make the changes.

**Comment by ericpassmore** [comment]
Note:start
category: System Stability
component: Internal
summary: Enable tracked_votes processing only if http chain_ro category is configured.
Note:end

File: plugins/chain_plugin/chain_plugin.cpp (hunk 1 above)
Comment (linh2931) [reply plugins/chain_plugin/chain_plugin.cpp:987]: Although more files are touched, the new approach hides the details of how http categories are intialized.

**Review by heifner — APPROVED** [review]

**Pull request merged**